src/main/java/carpet/mixins/Entity_scarpetEventsMixin.java

```diff
@@ -79,4 +84,19 @@ private void onRemove(CallbackInfo ci)
     }
 
 
+    @Inject(method = "setPos", at = @At("HEAD"))
+    private void firstPos(CallbackInfo ci)
+    {
+        pos1 = this.pos;
+        motion = this.velocity;
+    }
+
+    @Inject(method = "setPos", at = @At("TAIL"))
+    private void secondPos(CallbackInfo ci)
+    {
+        events.onEvent(EntityEventsGroup.Event.ON_MOVE,
+        motion.x, motion.y, motion.z,
+        pos1.x, pos1.y, pos1.z,
+        this.pos.x, this.pos.y, this.pos.z);
```

Comment:
You could also pass the `Vec3d`s to `ValueConversions.of(Vec3d)` instead of passing every single coordinate separately.

Reply:
Ah that's really convenient, I didn't know about `ValueConversions.of`!